src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -323,13 +324,21 @@ read_superblock(seastar::file &device, seastar::stat_data sd)
       bl.push_back(bp);
       block_sm_superblock_t ret;
       auto bliter = bl.cbegin();
+      // Validate the magic prefix
+      std::string sb_magic;
+      bliter.copy(SUPERBLOCK_MAGIC_LEN, sb_magic);
+      if (sb_magic != SUPERBLOCK_MAGIC) {
+        ERROR("invalid superblock magic: got '{}'", sb_magic);
```

Comment:
Probably good to also add "expected: '{}'" -> SUPERBLOCK_MAGIC

Reply:
done